dvc/stage.py

```diff
@@ -486,19 +486,24 @@ def _check_dvc_filename(fname):
             )
 
     @staticmethod
-    def _check_file_exists(fname):
-        if not os.path.exists(fname):
+    def _check_file_exists(repo, fname, branch):
+        if not repo.scm.exists(fname, branch):
             raise StageFileDoesNotExistError(fname)
 
     @staticmethod
-    def load(repo, fname):
-        Stage._check_file_exists(fname)
+    def load(repo, fname, branch=None):
+        # raise proper exceptions
```

Comment:
is it TODO, XXX? a better explanation is required for this comment ... in general, I don't like put TODOs and XXX into code as notifications .. usually people just don't care to fix later. I would go and create an RFC issue on github -  we'll have more chances to fix it, write tests, etc and keep code cleaner. But you should discuss this with @efiop and other guys.

Reply:
This was meant to be just a simple comment explaining the next two calls (I thought that they duplicate some checks down the function, but I don't see it now).